chef/cookbooks/crowbar/templates/default/crowbar-evacuate-host.sh.erb

```diff
@@ -41,17 +41,20 @@ blockmigrate=""
 <% end %>
 if ! nova host-evacuate-live "$host" $blockmigrate; then
     log "Live migration of instances from host $host has failed!"
-    touch $UPGRADEDIR/crowbar-evacuate-host-failed
+    echo 2 > $UPGRADEDIR/crowbar-evacuate-host-failed
     exit 2
 fi
 
-# FIXME: here we need checks for potential errors during the evacuation
-
 while nova list --all-tenants --host "$host" --fields host,status 2>/dev/null | grep -e ACTIVE -e MIGRATING | grep -q "$host"; do
     log "There is still some VM running at $host..."
     sleep 10
 done
 
+if nova list --all-tenants --host "$host" --fields status | grep -q ERROR; then
```

Comment:
I just tested and if the host which the instance is moving to suddenly dies, it will keep on migrating both on task_state and status until that node is back. If the destination node is not available then it will return to ACTIVE status on the original node.

So it does not seem like the instance will ever reach ERROR, which kind of makes sense as you dont want an instance which failed to migrate to be on that status, it will fail the migration but not error out.

My original thought was more regarding already existing instances on the host which are on error status and how this could be wrongly reported to the customer. Maybe they tried to launch 15 instances and 4 of them failed, they never cleaned them up.

So with this actual check those instances will be detected and reported as "Error during live-migration" which is not true.

Maybe we should skip this check?